spec/lib/omniauth_keycloak/configuration_spec.rb

```diff
@@ -0,0 +1,28 @@
+require 'spec_helper'
+
+describe 'OmniauthKeycloak::Configuration' do
+
+  describe 'Keycloak url detection' do
+    context 'without auth/ in url' do
+      before do
+        ENV["keycloak_url"] = 'http://auth-test.anynines.com/'
+      end
+
+      it 'should return correct url without auth/ in url' do
+        expect(OmniauthKeycloak::Configuration.new.realm_url).to eq('http://auth-test.anynines.com/realms/')
+      end
+    end 
+
+    context 'with auth/ in url' do
+      before do
+        ENV["keycloak_url"] = 'http://auth-test.anynines.com/'
+        ENV['keycloak_server_prefix'] ='auth'
+      end
```

Comment:
when settings envs, make sure to clean them up afterwards :) otherwise we run into undefined scenarios due to the random execution order of our tests.

either add an `after` block where you unset the envs, or make it sure that the env definitions in the spec helper are re-run after each test (basically reset it to the default, could be rspec before each test helper that sets the env vars back to the default)